### PR TITLE
reduce memory usage for centered rmsprop

### PR DESCRIPTION
**Pull request by meijieru**
Reduce gpu memory usage by using in-place operation

### CODE REVIEW

File: torch/optim/rmsprop.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def step(self, closure=None):
                 if group['centered']:
                     grad_avg = state['grad_avg']
                     grad_avg.mul_(alpha).add_(1 - alpha, grad)
-                    avg = square_avg.addcmul(-1, grad_avg, grad_avg).sqrt().add_(group['eps'])
+                    avg = square_avg.addcmul(-1, grad_avg, grad_avg).sqrt_().add_(group['eps'])
                 else:
                     avg = square_avg.sqrt().add_(group['eps'])
```
Comment (vincentqb) [inline torch/optim/rmsprop.py]: We can use `sqrt_` here too, no?

**Review by vincentqb — APPROVED** [review]
This looks good to me. I can merge after the comment is answered.

File: torch/optim/rmsprop.py (hunk 1 above)
Comment (meijieru) [reply torch/optim/rmsprop.py]: No. Since `square_avg` is a state that we maintain, we should not modify it here.

Comment (vincentqb) [reply torch/optim/rmsprop.py]: Right, good point.

**Pull request closed without merging**